src/test/kotlin/no/nb/mlt/wls/controller/ProductControllerTest.kt

```diff
@@ -0,0 +1,85 @@
+package no.nb.mlt.wls.controller
+
+import com.ninjasquad.springmockk.MockkBean
+import io.mockk.every
+import io.mockk.junit5.MockKExtension
+import no.nb.mlt.wls.EnableTestcontainers
+import no.nb.mlt.wls.core.data.Environment
+import no.nb.mlt.wls.core.data.HostName
+import no.nb.mlt.wls.core.data.Owner
+import no.nb.mlt.wls.core.data.Packaging
+import no.nb.mlt.wls.product.controller.ProductController
+import no.nb.mlt.wls.product.model.Product
+import no.nb.mlt.wls.product.repository.ProductRepository
+import no.nb.mlt.wls.product.service.ProductService
+import no.nb.mlt.wls.product.service.SynqService
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.data.mongodb.repository.config.EnableMongoRepositories
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.test.web.reactive.server.WebTestClient
+
+@EnableTestcontainers
+@SpringBootTest
+@EnableMongoRepositories("no.nb.mlt.wls.product.repository")
+@ExtendWith(MockKExtension::class)
+class ProductControllerTest {
+    @MockkBean
+    private lateinit var synqService: SynqService
+
+    @Autowired
+    lateinit var repository: ProductRepository
+
+    val testProduct =
+        Product(
+            hostName = HostName.AXIELL,
+            hostId = "mlt-2048",
+            productCategory = "BOOK",
+            description = "Ringenes Herre samling",
+            packaging = Packaging.BOX,
+            location = "SYNQ_WAREHOUSE",
+            quantity = 1.0,
```

Comment:
Quantity = 0 is a more appropriate value. We should consider either replacing whatever value we get for quantity and location in requests or throwing an error.